daemon/prompter.c

```diff
@@ -926,6 +927,27 @@ prompter_handle_invocation(prompter_t *self, GDBusMethodInvocation *invocation)
     prompter_eval_state_later(self);
 }
 
+void
+prompter_dbus_reload_config(prompter_t *self)
+{
+    log_info("reload dbus config");
+
+    if( prompter_is_connected(self) ) {
+        g_dbus_connection_call(prompter_connection(self),
```

Comment:
Note that session bus connection is owned by state machine and kept open only while there is something to prompt i.e. it is unlikely that active connection exists and this method call gets made. Given that we do not wait for reply and connect/disconnect is handled synchronously, one might get by with: check current connection status, connect if needed, do the method call and then close the connection if it was done for this purpose / stte.

Reply:
Done, it looked like this was working because I was until almost the last writing to ~/.local/share/dbus-1/services and that directory is watched I could see a reload happening after I thought I asked for one.

Closing the connection immediately after sending comes with a minor complication in the need to flush the connection first. I've done that asynchronously and closed the connection afterward but I may have over thought that.

Reply:
That async flush and disconnect looks nice. (In some old projects I've used dummy synchronous dbus daemon method calls as final step to ensure preceeding stuff has been pushed through, but that was with libdbus and its a bit iffy flush behavior.)